### PR TITLE
Added badge notice for `Atom-Ink` and added new badge spec

### PR DESCRIPTION
**Pull request by confused-Techie**
### Requirements 

* Filling out the template is required.
* All new code requires tests to ensure against regressions.
  - However, if your PR contains zero code changes, feel free to select the checkmark below to indicate so.

* [ ] Have you ran tests against this code?
* [X] This PR contains zero code changes.

### Description of the Change

This PR adds the badge addition to `atom-ink` to the admin actions log.

Additionally adding the definition of a new badge type `Broken`.

The need to add this badge was born from [this](https://github.com/pulsar-edit/pulsar/issues/482#issuecomment-1506542131) discussion.

### CODE REVIEW

File: docs/reference/badge-spec.md [hunk 1]
```diff
@@ -70,3 +70,11 @@ That is, if a Pulsar users on a supported platform, indepent of any other issues
...
+
+## Broken
+
+The `Broken` badge is used to indicate that the package available on the Pulsar Package Registry does not work at all in it's current form, on any supported platform. Either requiring manual changes to the source code, or otherwise being unrealistic to fix from the users perspective.
+
+This package likely emits warnings immediately, or may even cause the editor to crash as a whole. Installation of these packages is not recommended by the Pulsar team, and instead it is encouraged to work with the original maintainer to get these packages working, or otherwise the community is encouraged to maintain and manage a fork of said package.
+
+If a community member does decide to maintain a fork of a package with a `Broken` status, it's recommended to make the Pulsar Admins or Pulsar Backend Admins aware of this, so any warnings and links on the original package can be changed to recommend installation of your functional fork.
```
Comment (Daeraxa) [inline docs/reference/badge-spec.md:80]: One small comment is that we don't really define `Pulsar Admins or Pulsar Backend Admins` anywhere, maybe we should just make this a reference to the `Pulsar Team` as above or otherwise recommend something like creating an issue in the backend repo?

Comment (confused-Techie) [reply docs/reference/badge-spec.md:80]: This is a good point. Might make the most sense to just refer to the Pulsar Team.

Since not sure making an issue is the right move, although considering how infrequent of an event it'll be might not be that problematic.

**Comment by goyalyashpal** [comment]
I think the atom-ink package is likely on verge of being fixed. 

And also, no-one likely installs `atom-ink` themselves, it gets installed while installing `uber-juno`  package, so ...

Either way (be it fixed, or not), i think badging `atom-ink` won't be a very good idea.... if anything, uber-juno package should get a badge meanwhile due to its error with `Juno Error: Installing julia-client@0.12.6 failed` as shown in the image at https://github.com/pulsar-edit/pulsar/issues/482#issuecomment-1506577042

**Comment by confused-Techie** [comment]
@goyalyashpal I appreciate you contributing on this issue, so we may want to add it to both then, in the unlikely event that someone installs atom-ink` directly.

But appreciate the insight, and do hope that it is properly fixed sometime soon, so we will absolutely keep an eye out for it being done.

But otherwise thanks a ton for your feedback, and I'll edit this when I have the time

File: docs/reference/Admin_Actions.md [hunk 2]
```diff
@@ -4,6 +4,30 @@ When you consider that most backend services are a black box of code and decisio
 
 With that said this document will serve as the ongoing history of administrative actions that must be taken against the backend.
 
+## 2023 - April 15
+
+### Uber-Juno
+
+The [Uber-Juno](https://github.com/JunoLab/uber-juno) package uses an older unsupported coding style, sometimes called JavaScript Sloppy Mode, that's no longer supported in newer versions of NodeJS. Meaning the package crashes as soon as installed, and cannot be used without the user manually editing the source code.
```
Comment (goyalyashpal) [inline docs/reference/Admin_Actions.md:11]: as per the observation by comment https://github.com/pulsar-edit/pulsar/issues/482#issuecomment-1506542131 , it said in that error the following ...

> Modules are automatically in strict mode.

and since no issues are shown after that change, I think "it uses [Sloppy Mode](https://developer.mozilla.org/en-US/docs/Glossary/Sloppy_mode)" no longer applies...

File: docs/reference/Admin_Actions.md [hunk 3]
```diff
@@ -4,6 +4,30 @@ When you consider that most backend services are a black box of code and decisio
...
+
+One of the Pulsar contributors has [submitted a fix](https://github.com/JunoLab/uber-juno/pull/85) for this issue, although it has not been updated on the Pulsar Package Registry.
```
Comment (goyalyashpal) [inline docs/reference/Admin_Actions.md:13]:
```suggestion
[A fix](https://github.com/JunoLab/uber-juno/pull/85) from one of the Pulsar contributors has been merged, although it has not been updated on the Pulsar Package Registry.
```

Owing to the mention of "was accepted" in the section for Hydrogen (L22 in the master version)

File: docs/reference/Admin_Actions.md [hunk 4]
```diff
@@ -4,6 +4,30 @@ When you consider that most backend services are a black box of code and decisio
...
+
+Additionally, Uber-Juno relies on Atom-Ink, which is also broken without any fixes, as mentioned below.
+
+Due to these issues of the package, while the `main` branch is fixed, but it relies on broken dependencies, that render's the package as a whole still broken, and in which case will receive a [`broken`](./badge-spec.md#broken) badge.
```
Comment (goyalyashpal) [inline docs/reference/Admin_Actions.md:17]: Atom-Ink is also fixed.. so, this reason of "broken due to Atom-Ink" will have to be removed, and whole paragraph will have to be reworded to mention that issue with downloading as i mentioned in my previous comment:

`Juno Error: Installing julia-client@0.12.6 failed` as shown in the image at https://github.com/pulsar-edit/pulsar/issues/482#issuecomment-1506577042

File: docs/reference/Admin_Actions.md [hunk 5]
```diff
@@ -4,6 +4,30 @@ When you consider that most backend services are a black box of code and decisio
...
+
+### Atom-Ink
+
+The [Atom-Ink](https://github.com/JunoLab/atom-ink) package has many coding practices that are no longer valid in JavaScript/NodeJS, such as using reserved words, or using variables without declaring them first, causing the package to crash without manually being edited on Pulsar.
+
+One of the Pulsar contributors has [submitted a fix](https://github.com/JunoLab/atom-ink/pull/289) for this bug, but after four months of inactivity from the maintainer of the package it seems unlikely this will be resolved.
+
+The package is MIT licensed, so if any devs out there would like to pick up the reigns, and get this package functional as a fork, please feel free to do so, and you'd be encouraged to contact the Pulsar Admins about this, so we can update the badges on this package appropriately.
+
+It is for this reason it is **not** recommended to install the `atom-ink` package. It will not work, unless you are comfortable editing the source code of the package manually, or until a community member decides to maintain a fork of said package, which in that event, installing that package would be the recommendation here, rather than not installing.
```
Comment (goyalyashpal) [inline docs/reference/Admin_Actions.md:27]: This whole will have to be reworded as well... as the issue with using reserved words was with `uber-juno`, fix for undeclared variable use has been merged.


smth similar to instructions in line 26-28 (section hydrogen) `pulsar -p install ...` can work but since this is a dependency for uber-juno, i don't know what complications regarding its install that way will be...

 also, the upstream hasn't tagged a release _yet_. so, the command will have to target a commit for the time being... 

i have nudged upstream to tag it, but ...

**Review by goyalyashpal — CHANGES_REQUESTED** [review]
After both the fixing PRs have been merged, some comments and change requests regarding that.

**Comment by Daeraxa** [comment]
@goyalyashpal has the maintainer actually committed to submitting it to Pulsar? Unless they do then we still have the issue of the "bad" version being in the backend and a badge will still need to be applied.

**Comment by confused-Techie** [comment]
@goyalyashpal I appreciate the review, although we will still probably need to say both packages are broken.

Since `uber-juno` is essentially a package that just installs other packages. Which one of those is `atom-ink`. So there is no way to install `uber-juno` without having `atom-ink` begin to throw errors, which while `uber-juno` itself isn't broken it is effectively broken, since there's no way to stop those errors from being thrown without major edits to the source code. Since until a newer version of `atom-ink` is published to the Pulsar Package Registry, `uber-juno` will still throw errors.

But since you've done a fantastic job up until now in getting these maintainers responsive to the issues at hand I'll reword this as recommended but may wait a bit before applying any badging to see if they are able to get newer versions submitted.

Thanks for keeping us up to date!

**Comment by confused-Techie** [comment]
Alright as it's been two weeks with no new version published by the maintainers of these packages, I'll go ahead and update this here.

Additionally, since the `uber-juno` package will still install the broken version of the `ink` package (even if ink has the fixes on the git repo), there's no command we can use to install both from the git repo, as `uber-juno` will install what's available of ink on the registry.

So with all of that said, both packages are broken for better or worse, although we could include instructions on how to install `ink` from the git repo only.

Appreciate all the effort @goyalyashpal has put into this one, sorry the maintainers haven't come through to get things updated.

**Comment by goyalyashpal** [comment]
> there's no command we can use to install both from the git repo, as `uber-juno` will install what's available of ink on the registry.

this pretty much much did not make sense to me back then. but now, after having a hands on experience with paclage management _as a dev_ - this becomes perfectly clear.

wow on how well the concepts translate as following

* basic deps solver: 
`node_npm@npmjs  ~~ python_pip@pypi`
* full fledged env man with adv. solvers: <tt>node_[pnpm]@npmjs ~~ [micromamba]@conda-forge</tt>

<details>
<summary>Offtopic</summary>

Convergent evolution, node's <tt>[pnpm]</tt> boasts similar things about env mgmt & deps resolution as the language agnostic <tt>[micromamba]</tt> too lol.

[pnpm]: https://pnpm.io
[micromamba]: https://mamba.rtfd.io

This 2nd option for python uses a different distribution channel, the `conda-forge`, which allowed me to cope with lack of ready build packages there.

**edit:**
`pnpm add <pkg>` uses same repo/registry as npm i.e. MS Github's [`npmjs`.com](https://npmjs.com). 
Ref: [pnpm.io/cli/add](https://pnpm.io/cli/add#install-from-npm-registry). 

So, there doesnt _seem_ to exists any commuity registry. 

And well, even if it did then; pulsar would still be limited to repos supported by pac-man it uses - the npm

<strike>
so food for my own thoughts: are there diff repos for node deps as well? maybe they can solve not this issue, but at least the #586

what repo does the pnpm use?
</strike>

</details>
